pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -646,6 +656,56 @@ func (r *controlPlaneInstanceReconciler) renderCharts(ctx context.Context) error
 	return nil
 }
 
+func (r *controlPlaneInstanceReconciler) validateManifests(ctx context.Context) error {
+	log := common.LogFromContext(ctx)
+	allErrors := []error{}
+	// validate resource namespaces
+	smmr := &v1.ServiceMeshMemberRoll{}
+	if err := r.Client.Get(context.TODO(), client.ObjectKey{Namespace: r.Instance.GetNamespace(), Name: common.MemberRollName}, smmr); err != nil {
+		if !apierrors.IsNotFound(err) {
+			// log error, but don't fail validation: we'll just assume that the control plane namespace is the only namespace for now
+			log.Error(err, "failed to retrieve SMMR for SMCP")
+			smmr = nil
+		}
+	}
+	meshNamespaces := common.GetMeshNamespaces(r.Instance, smmr)
+	for _, manifestList := range r.renderings {
+		for _, manifest := range manifestList {
+			obj := map[string]interface{}{}
+			err := yaml.Unmarshal([]byte(manifest.Content), &obj)
+			if err != nil || obj == nil {
+				continue
+			}
+			withinMesh := false
+
+			metadata, ok := obj["metadata"].(map[string]interface{})
+			if !ok {
+				// if it doesn't have a metadata section, ignore
+				continue
+			}
+			objNs, ok := metadata["namespace"].(string)
+			if !ok {
+				// if namespace is not set, ignore
+				continue
+			}
+
+			for _, ns := range meshNamespaces {
```

Comment:
Would it be easier to return the set and just use that?

Reply:
Good point